src/Build.OM.UnitTests/Construction/ProjectTargetElement_Tests.cs

```diff
@@ -339,6 +343,42 @@ public void SetReturns()
             Assert.True(project.HasUnsavedChanges);
         }
 
+        /// <summary>
+        /// Parse invalid property under target
+        /// </summary>
+        [Fact]
+        public void ReadInvalidPropertyUnderTarget()
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                ChangeWaves.ResetStateForTests();
+                env.SetEnvironmentVariable("MSBUILDDISABLEFEATURESFROMVERSION", ChangeWaves.Wave17_6.ToString());
+                BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
+
```

Comment:
You should have two versions of this test: one that sets this, and one that doesn't. you can do that by using an xunit "Theory":

```suggestion
        [Theory]
        [InlineData(true)]
        [InlineData(false)]
        public void ReadInvalidPropertyUnderTarget(bool enableNewBehavior)
        {
            using (TestEnvironment env = TestEnvironment.Create())
            {
                ChangeWaves.ResetStateForTests();
                
                if (enableNewBehavior)
                {
                    env.SetEnvironmentVariable("MSBUILDDISABLEFEATURESFROMVERSION", ChangeWaves.Wave17_6.ToString());
                    BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
                }

```